pkg/syncer/util.go

```diff
@@ -170,29 +171,55 @@ func getPVCKey(ctx context.Context, obj interface{}) (string, error) {
 }
 
 // HasMigratedToAnnotationUpdate returns true if the migrated-to annotation is found in the newer object
-func HasMigratedToAnnotationUpdate(ctx context.Context, prevAnnotations map[string]string, newAnnotations map[string]string) bool {
+func HasMigratedToAnnotationUpdate(ctx context.Context, prevAnnotations map[string]string, newAnnotations map[string]string, objectName string) bool {
 	log := logger.GetLogger(ctx)
 	// Checking if the migrated-to annotation is found in the newer object
 	if _, annMigratedToFound := newAnnotations[common.AnnMigratedTo]; annMigratedToFound {
 		if _, annMigratedToFound = prevAnnotations[common.AnnMigratedTo]; !annMigratedToFound {
-			log.Debugf("Received %v annotation update", common.AnnMigratedTo)
+			log.Debugf("Received %v annotation update for %q", common.AnnMigratedTo, objectName)
+			return true
+		}
+	}
+	log.Debugf("%v annotation not found for %q", common.AnnMigratedTo, objectName)
+	return false
+}
+
+// isValidvSphereVolumeClaim returns true if the given PVC metadata of a vSphere Volume (in-tree volume)
+// has migrated-to annotation on the PVC
+// or if the PVC was provisioned by CSI driver using in-tree storage class
+func isValidvSphereVolumeClaim(ctx context.Context, pvcMetadata metav1.ObjectMeta) bool {
```

Comment:
We have  same logic in `isValidvSphereVolumeClaim` and `isValidvSphereVolume`.
Can we keep one function as following?

```
func isValidVSphereVolume(ctx context.Context, objectMetadata *metav1.ObjectMeta) bool {
```

We can supply PVC or PV metadata depending on the need.

Reply:
The annotation check is different in PVC and PV. 
For PVC we need to check if "volume.beta.kubernetes.io/storage-provisioner" is valid and for PV we need to check if "pv.kubernetes.io/provisioned-by" is valid. I think having 2 util functions is better.
Let me know if this sounds good